`gcc-8` fails to pass `make test-all`

Here's [actual log](https://github.com/t-mat/xxHash/runs/3345251074#step:11:30).

```
gcc-8 -O3 -Wall -Wextra -Wconversion -Wcast-qual -Wcast-align -Wshadow -Wstrict-aliasing=1 -Wswitch-enum -Wdeclaration-after-statement -Wstrict-prototypes -Wundef -Wpointer-arith -Wformat-security -Wvla -Wformat=2 -Winit-self -Wfloat-equal -Wwrite-strings -Wredundant-decls -Wstrict-overflow=2 -DXXH_DEBUGLEVEL=1  -Werror -DXXH_INLINE_ALL cli/xxhsum.c -o xxhsum_inlinedXXH
In file included from cli/xsum_sanity_check.c:35,
                 from cli/xxhsum.c:41:
cli/xsum_sanity_check.c: In function ‘XSUM_sanityCheck’:
cli/../xxhash.h:4656:12: error: assuming pointer wraparound does not occur when comparing P +- C1 with P +- C2 [-Werror=strict-overflow]
         if (input+XXH3_INTERNALBUFFER_SIZE < bEnd) {
            ^
cli/../xxhash.h:4656:12: error: assuming pointer wraparound does not occur when comparing P +- C1 with P +- C2 [-Werror=strict-overflow]
         if (input+XXH3_INTERNALBUFFER_SIZE < bEnd) {
            ^
cc1: all warnings being treated as errors
```


(1) `make test-all` fails with `gcc-8`.  But `gcc-9` passes it.

```
CC=gcc-8 make clean test-all    # FAIL
CC=gcc-9 make clean test-all    # PASS
```

(2) It seems it depends on optimization level.

```
CC=gcc-8 CFLAGS='-Werror -O3' DEBUGFLAGS='-DXXH_DEBUGLEVEL=1' make clean xxhsum_inlinedXXH      # FAIL
CC=gcc-8 CFLAGS='-Werror -O2' DEBUGFLAGS='-DXXH_DEBUGLEVEL=1' make clean xxhsum_inlinedXXH      # FAIL
CC=gcc-8 CFLAGS='-Werror -O1' DEBUGFLAGS='-DXXH_DEBUGLEVEL=1' make clean xxhsum_inlinedXXH      # PASS
CC=gcc-8 CFLAGS='-Werror -O0' DEBUGFLAGS='-DXXH_DEBUGLEVEL=1' make clean xxhsum_inlinedXXH      # PASS
```

(3) This error may be related to #549.

--- a/xxhash.h
+++ b/xxhash.h
@@ -4653,7 +4653,7 @@ XXH3_update(XXH3_state_t* state,
         XXH_ASSERT(input < bEnd);
 
         /* Consume input by a multiple of internal buffer size */
-        if (input+XXH3_INTERNALBUFFER_SIZE < bEnd) {
+        if (bEnd - input > XXH3_INTERNALBUFFER_SIZE) {
             const xxh_u8* const limit = bEnd - XXH3_INTERNALBUFFER_SIZE;
             do {
                 XXH3_consumeStripes(state->acc,

